Documentation/BasicOperators.md

```diff
@@ -46,41 +46,416 @@ code-style will be used.
+sendNext(numbersSink, "3")    // prints "3"
+sendCompleted(numbersSink)
+```
+
+[Interactive visualisation of the `flatten(.Concat)` operator.](http://neilpa.me/rac-marbles/#concat)
+
 ### Switching
 
+The `.Latest` strategy forwards only values from the latest input `SignalProducer`.
+
+```Swift
+let (producerA, sinkA) = SignalProducer<String, NoError>.buffer(5)
+let (producerB, sinkB) = SignalProducer<String, NoError>.buffer(5)
+let (producerC, sinkC) = SignalProducer<String, NoError>.buffer(5)
+let (signal, sink) = SignalProducer<SignalProducer<String, NoError>, NoError>.buffer(5)
+
+signal |> flatten(FlattenStrategy.Latest) |> start(next: println)
+
+sendNext(sink, producerA)   // nothing printed
+sendNext(sinkC, "X")        // nothing printed
+sendNext(sinkA, "a")        // prints "a"
+sendNext(sinkB, "1")        // nothing printed
+sendNext(sink, producerB)   // prints "1"
+sendNext(sinkA, "b")        // nothing printed
+sendNext(sinkB, "2")        // prints "2"
+sendNext(sinkC, "Y")        // nothing printed
+sendNext(sinkA, "c")        // nothing printed
+sendNext(sink, producerC)   // prints "X", "Y"
+sendNext(sinkB, "3")        // nothing printed
+sendNext(sinkC, "Z")        // prints "Z"
+```
 
 ## Handling errors
 
+These operators are used to handle errors that might occur on a signal.
+
 ### Catching errors
-### Mapping errors
+
+The `catch` operator catches any error that may occur on the input `SignalProducer`, then starts a new `SignalProducer` in its place.
+
+```Swift
+let (producer, sink) = SignalProducer<String, NSError>.buffer(5)
+let error = NSError(domain: "domain", code: 0, userInfo: nil)
+
+producer
+    |> catch { error in SignalProducer<String, NSError>(value: "Default") }
+    |> start(next: println)
+
+
+sendNext(sink, "First")     // prints "First"
+sendNext(sink, "Second")    // prints "Second"
+sendError(sink, error)      // prints "Default"
+```
+
 ### Retrying
 
+The `retry` operator will restart the original `SignalProducer` on error up to `count` times.
+
+```Swift
+var tries = 0
+let limit = 2
+let error = NSError(domain: "domain", code: 0, userInfo: nil)
+let producer = SignalProducer<String, NSError> { (sink, _) in
+    if tries++ < limit {
+        sendError(sink, error)
+    } else {
+        sendNext(sink, "Success")
+        sendCompleted(sink)
+    }
+}
+
+producer
+    |> on(error: {e in println("Error")})             // prints "Error" twice
+    |> retry(2)
+    |> start(next: println,                           // prints "Success"
+            error: { _ in println("Signal Error")})
+```
+
+If the `SignalProducer` does not succeed after `count` tries, the resulting `SignalProducer` will fail. E.g., if  `retry(1)` is used in the example above instead of `retry(2)`, `"Signal Error"` will be printed instead of `"Success"`.
+
+### Mapping errors
+
+The `mapError` operator transforms errors in the signal to new errors. 
+
+```Swift
+struct CustomError: ErrorType {
```

Comment:
:+1:  I had to add a few explicit type annotations to make the example run in the playground.